### ISSUE TITLE
--path-params-as-types doesn't work

### ISSUE THREAD

**Issue by shripadk** [openapi-ts/openapi-typescript#1003]
**Description**

Firstly, big thanks for creating this project @drwpow! Solved lot of my pain points.

`--path-params-as-types` doesn't work because `pathItemObject` in `transformPathsObject` is an object made up of HTTP methods (`get`, `post` etc). However, the code directly references [`pathItemObject.parameters`](https://github.com/drwpow/openapi-typescript/blob/903fd34729ae31c9a1d9efb55574d8b15abf4232/src/transform/paths-object.ts#L14) instead of looping through all method keys and then referencing `parameters` within it. Hence that block will always evaluate to false.

Reporting it here for now. Will open a PR if you are not taking it up or are busy with other stuff. So please let me know.

| Name                 | Version                      |
|:---------------------|:-----------------------------|
| `openapi-typescript` | `6.0.3`                      |
| Node.js              | `16.14.1`                      |
| OS + version         | `macOS 12.3.1` |

**Expected result**

Should generate dynamic string lookups

**Checklist**

- [x] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/CONTRIBUTING.md))

**Comment by mauriceoegerli** [comment]
I ran into this issue aswell. 
When debugging, i found out that in the file [paths-object.js](https://github.com/drwpow/openapi-typescript/blob/main/src/transform/paths-object.ts) there is a check for pathItemObject.parameters.
This object pathItemObject however never has this property, it is (from my testing) always nested unter the operation. In this case GET.
![image](https://user-images.githubusercontent.com/15955392/205058386-d1cd1d33-0f03-4e3e-bb6e-63a54785159d.png)

**Comment by steevsachs** [comment]
@shripadk Just ran into this today--thank you for diagnosing and raising the issue! Are you planning on opening a PR to resolve this?

**Comment by acnebs** [comment]
Yep AFAIK this was introduced in the v6 re-write.  @drwpow is --path-params-as-types working for you since v6?

**Comment by larkonesvk** [comment]
Same here, --path-params-as-types is not working.
I tried latest version, and 5.4.0 as well.

**Comment by mitchell-merry** [comment]
Can somebody please provide expected vs actual output here?

**Comment by felixfbecker** [comment]
I can confirm `--path-params-as-types` is not working in latest 6.2.4 but is working in 5.4.1.

In 6.2.4 the option has no effect, i.e. it will generate a plain string path:

```ts
interface paths {
  "/users/{id}": {
    // ...
  }
  "/posts/{id}" {
    // ...
  }
}
```

In 5.4.1 it generates a template literal string with the type of the parameter:

```ts
interface paths {
  [key: `/users/${number}`]: {
    // ...
  }
  [key: `/posts/${number}`] {
    // ...
  }
}
```

**Comment by felixfbecker** [comment]
If anyone is looking for a workaround, you can use this mapped type:

```ts
import { paths } from './generated-types.ts'

// Workaround for https://github.com/drwpow/openapi-typescript/issues/1003
type PathPattern<Pattern extends string, Params> = Pattern extends `${infer Head}/${infer Rest}`
  ? // Handle recursion
    `${PathPattern<Head, Params>}/${PathPattern<Rest, Params>}`
  : // Single segment
  Pattern extends `{${infer ParamName}}`
  ? // It's a param path segment
    Params extends { [_ in ParamName]: infer ParamType extends string | number }
    ? `${ParamType}`
    : string
  : // It's a literal path segment
    Pattern

type templatePaths = {
  [P in keyof paths as PathPattern<
    P,
    paths[P][keyof paths[P]] extends { parameters: { path: infer P } } ? P : never
  >]: paths[P]
}
```

Of course it would be a lot more efficient if we could codegen this directly, I have no idea what this may do to compile speeds (or if I have bugs in there).

**Comment by ElForastero** [comment]
If you use latest v5 version, you can patch a package, as an option.

Replace this:

`node_modules/openapi-typescript/dist/transform/paths.js`

```js
if (pathItem.parameters) {
    params = pathItem.parameters;
}
```

With this

```js
if (pathItem.parameters || pathItem.get?.parameters) {
    params = pathItem.parameters ?? pathItem.get.parameters;
}
```

**Comment by drwpow** [comment]
Semi-related: there were some bugs that were fixed in parameter generation (#1061) that may have been causing this.

But it’s also worth pointing out that this flag has always had caveats; there are instances where it can break. `openapi-fetch` doesn’t use this FWIW.

This flag probably needs some more robust tests to catch some of the errors outlined here. Would love PRs from any users relying on this feature! 🙏

**Comment by Hiieu** [comment]
This flag doesn't work in version `6.7.5`. When I run
```
npx openapi-typescript --path-params-as-types http://localhost:8000/api/v1/schema -o types/openapi-schema.d.ts
```
It hangs forever. If I remove the flag, then it generates the file, however, without dynamic lookups  :(
